// @aws-sdk/client-ssm/dist-es/commands/CreateResourceDataSyncCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateResourceDataSyncRequestFilterSensitiveLog, CreateResourceDataSyncResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1CreateResourceDataSyncCommand, serializeAws_json1_1CreateResourceDataSyncCommand, } from "../protocols/Aws_json1_1";
export class CreateResourceDataSyncCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "CreateResourceDataSyncCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: CreateResourceDataSyncRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateResourceDataSyncResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1CreateResourceDataSyncCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1CreateResourceDataSyncCommand(output, context);
    }
}
