// @aws-sdk/client-ssm/dist-es/commands/DeleteMaintenanceWindowCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteMaintenanceWindowRequestFilterSensitiveLog, DeleteMaintenanceWindowResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1DeleteMaintenanceWindowCommand, serializeAws_json1_1DeleteMaintenanceWindowCommand, } from "../protocols/Aws_json1_1";
export class DeleteMaintenanceWindowCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "DeleteMaintenanceWindowCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeleteMaintenanceWindowRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteMaintenanceWindowResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1DeleteMaintenanceWindowCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1DeleteMaintenanceWindowCommand(output, context);
    }
}
