// @aws-sdk/client-ssm/dist-es/commands/ListDocumentMetadataHistoryCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListDocumentMetadataHistoryRequestFilterSensitiveLog, ListDocumentMetadataHistoryResponseFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_json1_1ListDocumentMetadataHistoryCommand, serializeAws_json1_1ListDocumentMetadataHistoryCommand, } from "../protocols/Aws_json1_1";
export class ListDocumentMetadataHistoryCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "ListDocumentMetadataHistoryCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListDocumentMetadataHistoryRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListDocumentMetadataHistoryResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1ListDocumentMetadataHistoryCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1ListDocumentMetadataHistoryCommand(output, context);
    }
}
