// @aws-sdk/client-ssm/dist-es/commands/ListResourceDataSyncCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListResourceDataSyncRequestFilterSensitiveLog, ListResourceDataSyncResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_json1_1ListResourceDataSyncCommand, serializeAws_json1_1ListResourceDataSyncCommand, } from "../protocols/Aws_json1_1";
export class ListResourceDataSyncCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "ListResourceDataSyncCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListResourceDataSyncRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListResourceDataSyncResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1ListResourceDataSyncCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1ListResourceDataSyncCommand(output, context);
    }
}
