// @aws-sdk/client-ssm/dist-es/commands/ListOpsItemRelatedItemsCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListOpsItemRelatedItemsRequestFilterSensitiveLog, ListOpsItemRelatedItemsResponseFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_json1_1ListOpsItemRelatedItemsCommand, serializeAws_json1_1ListOpsItemRelatedItemsCommand, } from "../protocols/Aws_json1_1";
export class ListOpsItemRelatedItemsCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "ListOpsItemRelatedItemsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListOpsItemRelatedItemsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListOpsItemRelatedItemsResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1ListOpsItemRelatedItemsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1ListOpsItemRelatedItemsCommand(output, context);
    }
}
