// @aws-sdk/client-ssm/dist-es/commands/SendAutomationSignalCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SendAutomationSignalRequestFilterSensitiveLog, SendAutomationSignalResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_json1_1SendAutomationSignalCommand, serializeAws_json1_1SendAutomationSignalCommand, } from "../protocols/Aws_json1_1";
export class SendAutomationSignalCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "SendAutomationSignalCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: SendAutomationSignalRequestFilterSensitiveLog,
            outputFilterSensitiveLog: SendAutomationSignalResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1SendAutomationSignalCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1SendAutomationSignalCommand(output, context);
    }
}
