// @aws-sdk/client-ssm/dist-es/commands/StartChangeRequestExecutionCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartChangeRequestExecutionRequestFilterSensitiveLog, StartChangeRequestExecutionResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_json1_1StartChangeRequestExecutionCommand, serializeAws_json1_1StartChangeRequestExecutionCommand, } from "../protocols/Aws_json1_1";
export class StartChangeRequestExecutionCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "StartChangeRequestExecutionCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: StartChangeRequestExecutionRequestFilterSensitiveLog,
            outputFilterSensitiveLog: StartChangeRequestExecutionResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1StartChangeRequestExecutionCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1StartChangeRequestExecutionCommand(output, context);
    }
}
