// @aws-sdk/client-ssm/dist-es/commands/UpdateDocumentMetadataCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateDocumentMetadataRequestFilterSensitiveLog, UpdateDocumentMetadataResponseFilterSensitiveLog, } from "../models/models_2";
import { deserializeAws_json1_1UpdateDocumentMetadataCommand, serializeAws_json1_1UpdateDocumentMetadataCommand, } from "../protocols/Aws_json1_1";
export class UpdateDocumentMetadataCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "UpdateDocumentMetadataCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: UpdateDocumentMetadataRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UpdateDocumentMetadataResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1UpdateDocumentMetadataCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1UpdateDocumentMetadataCommand(output, context);
    }
}
