// @aws-sdk/client-ssm/dist-es/commands/UpdateManagedInstanceRoleCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateManagedInstanceRoleRequestFilterSensitiveLog, UpdateManagedInstanceRoleResultFilterSensitiveLog, } from "../models/models_2";
import { deserializeAws_json1_1UpdateManagedInstanceRoleCommand, serializeAws_json1_1UpdateManagedInstanceRoleCommand, } from "../protocols/Aws_json1_1";
export class UpdateManagedInstanceRoleCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "UpdateManagedInstanceRoleCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: UpdateManagedInstanceRoleRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UpdateManagedInstanceRoleResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1UpdateManagedInstanceRoleCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1UpdateManagedInstanceRoleCommand(output, context);
    }
}
