// @aws-sdk/client-ssm/dist-es/commands/UpdateMaintenanceWindowTargetCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateMaintenanceWindowTargetRequestFilterSensitiveLog, UpdateMaintenanceWindowTargetResultFilterSensitiveLog, } from "../models/models_2";
import { deserializeAws_json1_1UpdateMaintenanceWindowTargetCommand, serializeAws_json1_1UpdateMaintenanceWindowTargetCommand, } from "../protocols/Aws_json1_1";
export class UpdateMaintenanceWindowTargetCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SSMClient";
        const commandName = "UpdateMaintenanceWindowTargetCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: UpdateMaintenanceWindowTargetRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UpdateMaintenanceWindowTargetResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1UpdateMaintenanceWindowTargetCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1UpdateMaintenanceWindowTargetCommand(output, context);
    }
}
