// @aws-sdk/client-ssm/dist-es/models/models_2.js
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { PatchSourceFilterSensitiveLog, } from "./models_0";
import { MaintenanceWindowTaskInvocationParametersFilterSensitiveLog, } from "./models_1";
import { SSMServiceException as __BaseException } from "./SSMServiceException";
export var DocumentReviewAction;
(function (DocumentReviewAction) {
    DocumentReviewAction["Approve"] = "Approve";
    DocumentReviewAction["Reject"] = "Reject";
    DocumentReviewAction["SendForReview"] = "SendForReview";
    DocumentReviewAction["UpdateReview"] = "UpdateReview";
})(DocumentReviewAction || (DocumentReviewAction = {}));
export class OpsMetadataKeyLimitExceededException extends __BaseException {
    constructor(opts) {
        super({
            name: "OpsMetadataKeyLimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "OpsMetadataKeyLimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, OpsMetadataKeyLimitExceededException.prototype);
    }
}
export class ResourceDataSyncConflictException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceDataSyncConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceDataSyncConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceDataSyncConflictException.prototype);
        this.Message = opts.Message;
    }
}
export const UpdateDocumentDefaultVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentDefaultVersionDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateDocumentDefaultVersionResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DocumentReviewsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateDocumentMetadataRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateDocumentMetadataResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateMaintenanceWindowRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Description && { Description: SENSITIVE_STRING }),
});
export const UpdateMaintenanceWindowResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Description && { Description: SENSITIVE_STRING }),
});
export const UpdateMaintenanceWindowTargetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.OwnerInformation && { OwnerInformation: SENSITIVE_STRING }),
    ...(obj.Description && { Description: SENSITIVE_STRING }),
});
export const UpdateMaintenanceWindowTargetResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.OwnerInformation && { OwnerInformation: SENSITIVE_STRING }),
    ...(obj.Description && { Description: SENSITIVE_STRING }),
});
export const UpdateMaintenanceWindowTaskRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.TaskParameters && { TaskParameters: SENSITIVE_STRING }),
    ...(obj.TaskInvocationParameters && {
        TaskInvocationParameters: MaintenanceWindowTaskInvocationParametersFilterSensitiveLog(obj.TaskInvocationParameters),
    }),
    ...(obj.Description && { Description: SENSITIVE_STRING }),
});
export const UpdateMaintenanceWindowTaskResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.TaskParameters && { TaskParameters: SENSITIVE_STRING }),
    ...(obj.TaskInvocationParameters && {
        TaskInvocationParameters: MaintenanceWindowTaskInvocationParametersFilterSensitiveLog(obj.TaskInvocationParameters),
    }),
    ...(obj.Description && { Description: SENSITIVE_STRING }),
});
export const UpdateManagedInstanceRoleRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateManagedInstanceRoleResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateOpsItemRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateOpsItemResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateOpsMetadataRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateOpsMetadataResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdatePatchBaselineRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Sources && { Sources: obj.Sources.map((item) => PatchSourceFilterSensitiveLog(item)) }),
});
export const UpdatePatchBaselineResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Sources && { Sources: obj.Sources.map((item) => PatchSourceFilterSensitiveLog(item)) }),
});
export const UpdateResourceDataSyncRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateResourceDataSyncResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateServiceSettingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateServiceSettingResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const InventoryAggregatorFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OpsAggregatorFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GetInventoryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GetOpsSummaryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
