// @aws-sdk/client-sqs/dist-es/commands/AddPermissionCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AddPermissionRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_queryAddPermissionCommand, serializeAws_queryAddPermissionCommand, } from "../protocols/Aws_query";
export class AddPermissionCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SQSClient";
        const commandName = "AddPermissionCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: AddPermissionRequestFilterSensitiveLog,
            outputFilterSensitiveLog: (output) => output,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryAddPermissionCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryAddPermissionCommand(output, context);
    }
}
