// @aws-sdk/client-sqs/dist-es/commands/DeleteMessageBatchCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteMessageBatchRequestFilterSensitiveLog, DeleteMessageBatchResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDeleteMessageBatchCommand, serializeAws_queryDeleteMessageBatchCommand, } from "../protocols/Aws_query";
export class DeleteMessageBatchCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SQSClient";
        const commandName = "DeleteMessageBatchCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeleteMessageBatchRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteMessageBatchResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryDeleteMessageBatchCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryDeleteMessageBatchCommand(output, context);
    }
}
