// @aws-sdk/client-sqs/dist-es/commands/ListQueuesCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListQueuesRequestFilterSensitiveLog, ListQueuesResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListQueuesCommand, serializeAws_queryListQueuesCommand } from "../protocols/Aws_query";
export class ListQueuesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SQSClient";
        const commandName = "ListQueuesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListQueuesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListQueuesResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListQueuesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListQueuesCommand(output, context);
    }
}
