// @aws-sdk/client-sqs/dist-es/commands/ListDeadLetterSourceQueuesCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListDeadLetterSourceQueuesRequestFilterSensitiveLog, ListDeadLetterSourceQueuesResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListDeadLetterSourceQueuesCommand, serializeAws_queryListDeadLetterSourceQueuesCommand, } from "../protocols/Aws_query";
export class ListDeadLetterSourceQueuesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SQSClient";
        const commandName = "ListDeadLetterSourceQueuesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListDeadLetterSourceQueuesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListDeadLetterSourceQueuesResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListDeadLetterSourceQueuesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListDeadLetterSourceQueuesCommand(output, context);
    }
}
