// @aws-sdk/client-sqs/dist-es/commands/ListQueueTagsCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListQueueTagsRequestFilterSensitiveLog, ListQueueTagsResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListQueueTagsCommand, serializeAws_queryListQueueTagsCommand, } from "../protocols/Aws_query";
export class ListQueueTagsCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SQSClient";
        const commandName = "ListQueueTagsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListQueueTagsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListQueueTagsResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListQueueTagsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListQueueTagsCommand(output, context);
    }
}
