// @aws-sdk/client-sqs/dist-es/commands/ReceiveMessageCommand.js
import { getReceiveMessagePlugin } from "@aws-sdk/middleware-sdk-sqs";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ReceiveMessageRequestFilterSensitiveLog, ReceiveMessageResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryReceiveMessageCommand, serializeAws_queryReceiveMessageCommand, } from "../protocols/Aws_query";
export class ReceiveMessageCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getReceiveMessagePlugin(configuration));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SQSClient";
        const commandName = "ReceiveMessageCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ReceiveMessageRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ReceiveMessageResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryReceiveMessageCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryReceiveMessageCommand(output, context);
    }
}
