// @aws-sdk/client-sqs/dist-es/commands/SendMessageBatchCommand.js
import { getSendMessageBatchPlugin } from "@aws-sdk/middleware-sdk-sqs";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SendMessageBatchRequestFilterSensitiveLog, SendMessageBatchResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_querySendMessageBatchCommand, serializeAws_querySendMessageBatchCommand, } from "../protocols/Aws_query";
export class SendMessageBatchCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getSendMessageBatchPlugin(configuration));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SQSClient";
        const commandName = "SendMessageBatchCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: SendMessageBatchRequestFilterSensitiveLog,
            outputFilterSensitiveLog: SendMessageBatchResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_querySendMessageBatchCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_querySendMessageBatchCommand(output, context);
    }
}
