// @aws-sdk/client-apigatewaymanagementapi/dist-es/commands/DeleteConnectionCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteConnectionRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteConnectionCommand, serializeAws_restJson1DeleteConnectionCommand, } from "../protocols/Aws_restJson1";
export class DeleteConnectionCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ApiGatewayManagementApiClient";
        const commandName = "DeleteConnectionCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeleteConnectionRequestFilterSensitiveLog,
            outputFilterSensitiveLog: (output) => output,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restJson1DeleteConnectionCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restJson1DeleteConnectionCommand(output, context);
    }
}
