// @aws-sdk/client-dynamodb/dist-es/commands/BatchGetItemCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetItemInputFilterSensitiveLog, BatchGetItemOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_0BatchGetItemCommand, serializeAws_json1_0BatchGetItemCommand, } from "../protocols/Aws_json1_0";
export class BatchGetItemCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "DynamoDBClient";
        const commandName = "BatchGetItemCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: BatchGetItemInputFilterSensitiveLog,
            outputFilterSensitiveLog: BatchGetItemOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_0BatchGetItemCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_0BatchGetItemCommand(output, context);
    }
}
