// @aws-sdk/client-dynamodb/dist-es/commands/DescribeGlobalTableSettingsCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeGlobalTableSettingsInputFilterSensitiveLog, DescribeGlobalTableSettingsOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_0DescribeGlobalTableSettingsCommand, serializeAws_json1_0DescribeGlobalTableSettingsCommand, } from "../protocols/Aws_json1_0";
export class DescribeGlobalTableSettingsCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "DynamoDBClient";
        const commandName = "DescribeGlobalTableSettingsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DescribeGlobalTableSettingsInputFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeGlobalTableSettingsOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_0DescribeGlobalTableSettingsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_0DescribeGlobalTableSettingsCommand(output, context);
    }
}
