// @aws-sdk/client-sns/dist-es/commands/ConfirmSubscriptionCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ConfirmSubscriptionInputFilterSensitiveLog, ConfirmSubscriptionResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryConfirmSubscriptionCommand, serializeAws_queryConfirmSubscriptionCommand, } from "../protocols/Aws_query";
export class ConfirmSubscriptionCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "ConfirmSubscriptionCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ConfirmSubscriptionInputFilterSensitiveLog,
            outputFilterSensitiveLog: ConfirmSubscriptionResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryConfirmSubscriptionCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryConfirmSubscriptionCommand(output, context);
    }
}
