// @aws-sdk/client-sns/dist-es/commands/DeletePlatformApplicationCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeletePlatformApplicationInputFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_queryDeletePlatformApplicationCommand, serializeAws_queryDeletePlatformApplicationCommand, } from "../protocols/Aws_query";
export class DeletePlatformApplicationCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "DeletePlatformApplicationCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeletePlatformApplicationInputFilterSensitiveLog,
            outputFilterSensitiveLog: (output) => output,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryDeletePlatformApplicationCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryDeletePlatformApplicationCommand(output, context);
    }
}
