// @aws-sdk/client-sns/dist-es/commands/GetEndpointAttributesCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetEndpointAttributesInputFilterSensitiveLog, GetEndpointAttributesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryGetEndpointAttributesCommand, serializeAws_queryGetEndpointAttributesCommand, } from "../protocols/Aws_query";
export class GetEndpointAttributesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "GetEndpointAttributesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: GetEndpointAttributesInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetEndpointAttributesResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryGetEndpointAttributesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryGetEndpointAttributesCommand(output, context);
    }
}
