// @aws-sdk/client-sns/dist-es/commands/GetPlatformApplicationAttributesCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetPlatformApplicationAttributesInputFilterSensitiveLog, GetPlatformApplicationAttributesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryGetPlatformApplicationAttributesCommand, serializeAws_queryGetPlatformApplicationAttributesCommand, } from "../protocols/Aws_query";
export class GetPlatformApplicationAttributesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "GetPlatformApplicationAttributesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: GetPlatformApplicationAttributesInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetPlatformApplicationAttributesResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryGetPlatformApplicationAttributesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryGetPlatformApplicationAttributesCommand(output, context);
    }
}
