// @aws-sdk/client-sns/dist-es/commands/GetSMSAttributesCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetSMSAttributesInputFilterSensitiveLog, GetSMSAttributesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryGetSMSAttributesCommand, serializeAws_queryGetSMSAttributesCommand, } from "../protocols/Aws_query";
export class GetSMSAttributesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "GetSMSAttributesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: GetSMSAttributesInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetSMSAttributesResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryGetSMSAttributesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryGetSMSAttributesCommand(output, context);
    }
}
