// @aws-sdk/client-sns/dist-es/commands/ListSMSSandboxPhoneNumbersCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSMSSandboxPhoneNumbersInputFilterSensitiveLog, ListSMSSandboxPhoneNumbersResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListSMSSandboxPhoneNumbersCommand, serializeAws_queryListSMSSandboxPhoneNumbersCommand, } from "../protocols/Aws_query";
export class ListSMSSandboxPhoneNumbersCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "ListSMSSandboxPhoneNumbersCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListSMSSandboxPhoneNumbersInputFilterSensitiveLog,
            outputFilterSensitiveLog: ListSMSSandboxPhoneNumbersResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListSMSSandboxPhoneNumbersCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListSMSSandboxPhoneNumbersCommand(output, context);
    }
}
