// @aws-sdk/client-sns/dist-es/commands/ListSubscriptionsByTopicCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSubscriptionsByTopicInputFilterSensitiveLog, ListSubscriptionsByTopicResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListSubscriptionsByTopicCommand, serializeAws_queryListSubscriptionsByTopicCommand, } from "../protocols/Aws_query";
export class ListSubscriptionsByTopicCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "ListSubscriptionsByTopicCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListSubscriptionsByTopicInputFilterSensitiveLog,
            outputFilterSensitiveLog: ListSubscriptionsByTopicResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListSubscriptionsByTopicCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListSubscriptionsByTopicCommand(output, context);
    }
}
