// @aws-sdk/client-sns/dist-es/commands/ListOriginationNumbersCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListOriginationNumbersRequestFilterSensitiveLog, ListOriginationNumbersResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListOriginationNumbersCommand, serializeAws_queryListOriginationNumbersCommand, } from "../protocols/Aws_query";
export class ListOriginationNumbersCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "ListOriginationNumbersCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListOriginationNumbersRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListOriginationNumbersResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListOriginationNumbersCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListOriginationNumbersCommand(output, context);
    }
}
