// @aws-sdk/client-sns/dist-es/commands/ListPlatformApplicationsCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPlatformApplicationsInputFilterSensitiveLog, ListPlatformApplicationsResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryListPlatformApplicationsCommand, serializeAws_queryListPlatformApplicationsCommand, } from "../protocols/Aws_query";
export class ListPlatformApplicationsCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "ListPlatformApplicationsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListPlatformApplicationsInputFilterSensitiveLog,
            outputFilterSensitiveLog: ListPlatformApplicationsResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryListPlatformApplicationsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryListPlatformApplicationsCommand(output, context);
    }
}
