// @aws-sdk/client-sns/dist-es/commands/OptInPhoneNumberCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { OptInPhoneNumberInputFilterSensitiveLog, OptInPhoneNumberResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryOptInPhoneNumberCommand, serializeAws_queryOptInPhoneNumberCommand, } from "../protocols/Aws_query";
export class OptInPhoneNumberCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "OptInPhoneNumberCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: OptInPhoneNumberInputFilterSensitiveLog,
            outputFilterSensitiveLog: OptInPhoneNumberResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryOptInPhoneNumberCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryOptInPhoneNumberCommand(output, context);
    }
}
