// @aws-sdk/client-sns/dist-es/commands/PublishBatchCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PublishBatchInputFilterSensitiveLog, PublishBatchResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryPublishBatchCommand, serializeAws_queryPublishBatchCommand } from "../protocols/Aws_query";
export class PublishBatchCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "PublishBatchCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: PublishBatchInputFilterSensitiveLog,
            outputFilterSensitiveLog: PublishBatchResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryPublishBatchCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryPublishBatchCommand(output, context);
    }
}
