// @aws-sdk/client-sns/dist-es/commands/SetTopicAttributesCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SetTopicAttributesInputFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_querySetTopicAttributesCommand, serializeAws_querySetTopicAttributesCommand, } from "../protocols/Aws_query";
export class SetTopicAttributesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "SetTopicAttributesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: SetTopicAttributesInputFilterSensitiveLog,
            outputFilterSensitiveLog: (output) => output,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_querySetTopicAttributesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_querySetTopicAttributesCommand(output, context);
    }
}
