// @aws-sdk/client-sns/dist-es/commands/VerifySMSSandboxPhoneNumberCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { VerifySMSSandboxPhoneNumberInputFilterSensitiveLog, VerifySMSSandboxPhoneNumberResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryVerifySMSSandboxPhoneNumberCommand, serializeAws_queryVerifySMSSandboxPhoneNumberCommand, } from "../protocols/Aws_query";
export class VerifySMSSandboxPhoneNumberCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SNSClient";
        const commandName = "VerifySMSSandboxPhoneNumberCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: VerifySMSSandboxPhoneNumberInputFilterSensitiveLog,
            outputFilterSensitiveLog: VerifySMSSandboxPhoneNumberResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryVerifySMSSandboxPhoneNumberCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryVerifySMSSandboxPhoneNumberCommand(output, context);
    }
}
