// @aws-sdk/client-sqs/dist-cjs/commands/SendMessageCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SendMessageCommand = void 0;
const middleware_sdk_sqs_1 = require("@aws-sdk/middleware-sdk-sqs");
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_0_1 = require("../models/models_0");
const Aws_query_1 = require("../protocols/Aws_query");
class SendMessageCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use((0, middleware_serde_1.getSerdePlugin)(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use((0, middleware_sdk_sqs_1.getSendMessagePlugin)(configuration));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "SQSClient";
        const commandName = "SendMessageCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_0_1.SendMessageRequestFilterSensitiveLog,
            outputFilterSensitiveLog: models_0_1.SendMessageResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return (0, Aws_query_1.serializeAws_querySendMessageCommand)(input, context);
    }
    deserialize(output, context) {
        return (0, Aws_query_1.deserializeAws_querySendMessageCommand)(output, context);
    }
}
exports.SendMessageCommand = SendMessageCommand;
